### PR TITLE
refactor(icon): use design system icon names

### PR DESCRIPTION
**Pull request by jamime**
### Proposed behaviour
Refactor the icon unicodes to follow the convention on the design system so it is easier to maintain.

### Current behaviour
The file is hard to maintain with a lot of legacy/duplicates.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
I first added a test with a snapshot so we can be sure the final list of unicodes did not change.

Fixes FE-4293

### Testing instructions
Jest tests should be passing and there should be no difference to any component.

### CODE REVIEW

File: src/components/icon/__snapshots__/icon-unicodes.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,206 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`the unicodes have not changed 1`] = `
+Object {
+  "add": "\\\\e940",
```
Comment (dawidzarzycki) [inline src/components/icon/__snapshots__/icon-unicodes.spec.js.snap:5]: Is it correct that we have 4 `\` instead of 2?

Comment (edleeks87) [reply src/components/icon/__snapshots__/icon-unicodes.spec.js.snap:5]: It's adding escape characters which it doesn't seem to do in any of the snapshots for components using `Icon` 🤔

Comment (jamime) [reply src/components/icon/__snapshots__/icon-unicodes.spec.js.snap:5]: So we actually need `"\e940"` `\` by itself is an escape char, so we need a second one to cancel the escape and treat it as a normal backslash.

I'm not sure why jest adds additional slashes, two should be sufficient. My guess is they replace every `\` with two.
I logged out the object to check and the object does only contain two backslashes as expected.

image

**Review by edleeks87 — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]
no `chromatic` changes noticed

**Pull request merged**